doc/source/whatsnew/v2.1.0.rst

```diff
@@ -478,10 +478,10 @@ Conversion
 - Bug in :func:`DataFrame.style.to_latex` and :func:`DataFrame.style.to_html` if the DataFrame contains integers with more digits than can be represented by floating point double precision (:issue:`52272`)
 - Bug in :func:`array`  when given a ``datetime64`` or ``timedelta64`` dtype with unit of "s", "us", or "ms" returning :class:`PandasArray` instead of :class:`DatetimeArray` or :class:`TimedeltaArray` (:issue:`52859`)
 - Bug in :meth:`ArrowDtype.numpy_dtype` returning nanosecond units for non-nanosecond ``pyarrow.timestamp`` and ``pyarrow.duration`` types (:issue:`51800`)
+- Bug in :meth:`BinOp.convert_value` loses precision when converting int to float if integer in DataFrame stored in HDFS file format is large (:issue:`54186`)
```

Comment:
Could you rewrite this in terms of a public pandas API (i.e. BInOp is not public)

Reply:
Rewrote it referencing HDFStore select method.